Controller/Adminhtml/Menu/Delete.php

```diff
@@ -53,19 +81,22 @@ public function __construct(
     /**
      * Dispatch request
      *
-     * @return \Magento\Framework\Controller\ResultInterface|ResponseInterface
-     * @throws \Magento\Framework\Exception\NotFoundException
+     * @return ResultInterface|ResponseInterface
+     * @throws NotFoundException
      */
     public function execute()
     {
-        $id = $this->getRequest()->getParam('id');
+        $menuId = (int) $this->getRequest()->getParam('id');
 
         try {
-            $menu = $this->menuRepository->getById($id);
-            $this->menuRepository->deleteById($id);
+            $menu = $this->menuRepository->getById($menuId);
```

Comment:
`->getById($menuId)` throws a `NoSuchEntityException`, I wonder if we should catch it as well, same as with `CouldNotDeleteException`. I see both extend the `LocalizedException`, maybe catch this one only, what do you think?

Reply:
Added another line to catch `NoSuchEntityException` exception. I think this way more clear than using one `LocalizedException` exception.